### PR TITLE
migrate MyNativeView to Kotlin

### PR DESCRIPTION
**Pull request by BogiKay**
## Summary:

This PR converts MyNativeView into Kotlin as requested in https://github.com/facebook/react-native/issues/38825 .

## Changelog:

[INTERNAL] [CHANGED] - Migrate MyNativeView to Kotlin

## Test Plan:

1. run `yarn android`
2. Check whether RN Tester runs as expected

### CODE REVIEW

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  private fun emitNativeEvent(color: Int) {
+    val event = Arguments.createMap()
+    val backgroundColor = Arguments.createMap()
+    val hsv = FloatArray(3)
+
+    Color.colorToHSV(color, hsv)
+    backgroundColor.putDouble("hue", hsv[0].toDouble())
+    backgroundColor.putDouble("saturation", hsv[1].toDouble())
+    backgroundColor.putDouble("brightness", hsv[2].toDouble())
+    backgroundColor.putDouble("alpha", Color.alpha(color).toDouble())
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt:50]:
```suggestion
    val backgroundColor = Arguments.createMap().apply {
      putDouble("hue", hsv[0].toDouble())
      putDouble("saturation", hsv[1].toDouble())
      putDouble("brightness", hsv[2].toDouble())
      putDouble("alpha", Color.alpha(color).toDouble())
    }
```

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    payload.putArray("values", newIntArray)
+    payload.putArray("boolValues", newBoolArray)
+    payload.putArray("floats", newFloatArray)
+    payload.putArray("doubles", newDoubleArray)
+    payload.putArray("yesNos", newYesNoArray)
+    payload.putArray("strings", newStringArray)
+    payload.putArray("latLons", newObjectArray)
+    payload.putArray("multiArrays", newArrayArray)
```
Comment (cortinico) [inline packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt:95]: same as above

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for sending this over @BogiKay I've left a couple of comments and then we can merge this

File: packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt (hunk 1 above)
Comment (BogiKay) [reply packages/rn-tester/android/app/src/main/java/com/facebook/react/uiapp/component/MyNativeView.kt]: Thank you for your feedback. Code updated

**Review by mdvacca — APPROVED** [review]
lgtm

**Pull request closed without merging**